app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class FoodsController < ApplicationController
+  load_and_authorize_resource
+
+  def index
+    @current_user = current_user
+    @foods = current_user.foods.all.includes(:recipe_foods)
+  end
+
+  def new
+    @food = Food.new
+    @current_user = current_user
+  end
+
+  def create
+    @food = current_user.foods.new(food_params)
+    if @food.save
+      redirect_to foods_url, notice: 'Food was successfully created.'
+    else
+      flash[:alert] = 'Error: Food is not published'
+    end
+  end
+
+  def destroy
+    @food.destroy
+    redirect_to foods_url, notice: 'Food was successfully destroyed.'
+  end
```

Comment:
- [ ] when i tried to delete a food I had this error, kindly fix this 
![Screenshot from 2023-09-22 22-54-52](https://github.com/lorenzoworx/chef-chapters/assets/66097371/c8953bef-18da-4a8b-8b48-2d9a1a70a9db)